src/pyFAI/gui/dialog/DetectorSelectorDialog.py

```diff
@@ -143,50 +143,95 @@ def __init__(self, parent=None):
+
+            # Repopulate
+            if detector and detector.SENSORS:
+                materials = set()  # Avoid duplicated population
+                thicknesses = set()
+                for config in detector.SENSORS:
+                    info.append(str(config))
+                    mat = config.material
+                    thick = config.thickness
+                    if mat not in materials:
+                        self._detectorSensorMaterials.addItem(
+                            mat.name, userData=mat
+                        )
+                        materials.add(mat)
+                    if thick not in thicknesses:
+                        stg = f"{1e6 * thick:4.0f}" if thick else ""
+                        self._detectorSensorThickness.addItem(stg, userData=thick)
+                        thicknesses.add(thick)
+            else:
+                self._detectorSensorMaterials.addItem(" ", userData=None)
+                for key, value in detectors.sensors.ALL_MATERIALS.items():
+                    self._detectorSensorMaterials.addItem(key, userData=value)
+                self._detectorSensorThickness.addItem("", userData=None)
+            self._detectorSensorInfo.setText("|".join(info))
+
+        if isinstance(detector, detectors.Detector):
+            self.setSensorConfig(detector.sensor)
+
+    def getSensorConfig(self) -> SensorConfig:
+        sensor_material = self._detectorSensorMaterials.currentData()
+        if sensor_material:
+            sensor = SensorConfig(sensor_material)
+            thickness = self._detectorSensorThickness.currentText()  # Not data since can be user modified
             if thickness.strip():
-                sensor.thickness = 1e-6*float(thickness)
+                sensor.thickness = 1e-6 * float(thickness)
             else:
                 sensor.thickness = None
         else:
             sensor = None
         return sensor
 
-    def setSensorConfig(self, sensor=None):
-        if sensor:
-            thickness = f"{1e6*sensor.thickness:.1f}" or ""
-            self._detectorSensorThickness.setText(thickness)
-            index = self._detectorSensorMaterials.findData(sensor.material)
+    def setSensorConfig(self, sensor: SensorConfig | None = None):
+        if sensor is None:
+            sensor = SensorConfig(None, None)
+        index = self._detectorSensorThickness.findData(sensor.thickness)
+        with block_signals(self._detectorSensorThickness):
+            if index < 0:
+                self._detectorSensorThickness.addItem(
+                    f"{1e6 * sensor.thickness:4.0f}" if sensor.thickness else "", 
+                    userData=sensor.thickness
+                )
+                index = self._detectorSensorThickness.findData(sensor.thickness)
+            self._detectorSensorThickness.setCurrentIndex(index)
+
+        index = self._detectorSensorMaterials.findData(sensor.material)
+        with block_signals(self._detectorSensorMaterials):
```

Comment:
Is blocking signals really needed here?
Unless I missed something, when calling this method, `__sensorChanged` is not called, so the actual senor info is not updated....
Does it matter if `__sensorChanged` is called twice?

Reply:
I checked an `__sensorChanged` was called (too) many times.